server/plugins/renderer/responses/pencil-response.spec.js

```diff
@@ -0,0 +1,67 @@
+const Code = require('code');
+const Lab = require('@hapi/lab');
+const sinon = require('sinon');
+const PencilResponse = require('./pencil-response');
+const lab = exports.lab = Lab.script();
+const expect = Code.expect;
+const it = lab.it;
+
+
+lab.describe('PencilResponse', () => {
+    const assembler = {
+        getTemplates: path => new Promise(resolve => resolve({ path })),
+        getTranslations: () => new Promise(resolve => resolve([])),
+    };
+
+    let data;
+    let request;
+    let response;
+    let h;
+
+
+    lab.beforeEach(() => {
+        data = {
+            context: {
+                settings: {},
+                theme_settings: {},
+                template_engine: 'handlebars-v3',
+            },
+            headers: {},
+            template_file: null,
+            remote: true,
+            remote_data: '',
+        };
+    
+        request = {
+            url: {},
+            path: '/',
+            app: {themeConfig: {variationIndex: 1}},
+            headers: {},
+            query: {},
+        };
+
+        response = {
+            code: () => response,
+        };
+
+        h = {
+            response: sinon.stub().returns(response),
+        };
+
+    });
+
+    it('should return error, when the wrong template_engine is sent', () => {
```

Comment:
We need to test for 3 cases;

- When the key doesn't exist `->` In this case it should default to `handlebars-v3`
- When the key exists but it doesn't match a predefined list `->` It should throw an error
- When key exists and it matches predefined list `->` It should render using the passed template engine